pkg/devfile/parser/context/apiVersion.go

```diff
@@ -19,19 +19,34 @@ func (d *DevfileCtx) SetDevfileAPIVersion() error {
 		return errors.Wrapf(err, "failed to decode devfile json")
 	}
 
-	// Get "apiVersion" value from the map
-	apiVersion, ok := r["apiVersion"]
-	if !ok {
-		return fmt.Errorf("apiVersion not present in devfile")
-	}
+	var apiVer string
+
+	// Get "apiVersion" value from map for devfile V1
+	apiVersion, okApi := r["apiVersion"]
+
+	// Get "schemaVersion" value from map for devfile V2
+	schemaVersion, okSchema := r["schemaVersion"]
```

Comment:
what happens if both are provided, apiVersion and schemaVersion?

Reply:
there is no field `apiVersion` for V2 and no field `schemaVersion` for V1, so they cannot be provided together, json validation would fail for such case.